mdp/nodes/sfa_nodes.py

```diff
@@ -369,6 +376,170 @@ def get_eta_values(self, t=1):
+    the SFA problem formulation by applying the trapezoid rule.
+
+    .. attribute:: avg
+
+        Mean of the input data (available after training)
+
+    .. attribute:: sf
+
+        Matrix of the SFA filters (available after training)
+
+    .. attribute:: d
+
+        Delta values corresponding to the SFA components (generalized
+        eigenvalues). [See the docs of the ``get_eta_values`` method for
+        more information]
+
+    .. admonition:: Reference
+
+        More information about Slow Feature Analysis can be found in
+        Wiskott, L. and Sejnowski, T.J., Slow Feature Analysis: Unsupervised
+        Learning of Invariances, Neural Computation, 14(4):715-770 (2002).
+    """
+
+    def __init__(self, input_dim=None, output_dim=None, dtype=None,
+                 rank_deficit_method='none'):
+        """"
+        Initialize an object of type 'VartimeSFANode'.
+
+        :param input_dim: The input dimensionality.
+        :type input_dim: int
+
+        :param output_dim: The output dimensionality.
+        :type output_dim: int
+
+        :param dtype: The datatype.
+        :type dtype: numpy.dtype or str
+
+        :param rank_deficit_method: Possible values: 'none' (default), 'reg', 'pca', 'svd', 'auto'
+            If not 'none', the ``stop_train`` method solves the SFA eigenvalue
+            problem in a way that is robust against linear redundancies in
+            the input data. This would otherwise lead to rank deficit in the
+            covariance matrix, which usually yields a
+            SymeigException ('Covariance matrices may be singular').
+            There are several solving methods implemented:
+
+            reg  - works by regularization
+            pca  - works by PCA
+            svd  - works by SVD
+            ldl  - works by LDL decomposition (requires SciPy >= 1.0)
+
+            auto - (Will be: selects the best-benchmarked method of the above)
+                   Currently it simply selects pca.
+
+            Note: If you already received an exception
+            SymeigException ('Covariance matrices may be singular')
+            you can manually set the solving method for an existing node::
+
+               sfa.set_rank_deficit_method('pca')
+
+            That means,::
+
+               sfa = SFANode(rank_deficit='pca')
+
+            is equivalent to::
+
+               sfa = SFANode()
+               sfa.set_rank_deficit_method('pca')
+
+            After such an adjustment you can run ``stop_training()`` again,
+            which would save a potentially time-consuming rerun of all
+            ``train()`` calls.
+        :type rank_deficit_method: str
+        """
+        super(VartimeSFANode, self).__init__(input_dim,
+                                             output_dim, dtype, include_last_sample=True,
+                                             rank_deficit_method=rank_deficit_method)
+
+        self.tphase = 0
```

Comment:
It looks like you use tphases as chunks-counter. Please fix the clash between chunks and phases (also in the doc).